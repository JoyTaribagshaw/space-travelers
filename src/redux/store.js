import { configureStore } from '@reduxjs/toolkit';
import { logger } from 'redux-logger';
import rocketsReducer from './rockets/rocketsSlice';
import missionsReducer from './missions/missionsSlice';
import dragonsReducer from './dragons/dragonsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsReducer,
    dragons: dragonsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;