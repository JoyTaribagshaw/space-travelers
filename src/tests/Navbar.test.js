import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Navbar from '../Components/navigation';

test('Check if Navbar component is rendered correctly', () => {
  const tree = render(
    <Provider store={store}>
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    </Provider>,
  );

  expect(tree).toMatchSnapshot();
});
