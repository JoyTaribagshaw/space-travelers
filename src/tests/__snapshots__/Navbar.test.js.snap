// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Navbar component is rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="px-16 py-8 flex justify-between items-center border-b-2 text-blue-500 w-full"
      >
        <div
          class="flex items-center gap-4"
        >
          <img
            alt=""
            height="50"
            src="planet.png"
            width="50"
          />
          <h1
            class="text-black text-lg"
          >
            Space Travelers Hub
          </h1>
        </div>
        <ul
          class="flex gap-4 border-b-slate-400"
        >
          <li
            class="link"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li
            class="link"
          >
            <a
              class=""
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li
            class="link"
          >
            <a
              class=""
              href="/myprofile"
            >
              My profile
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="px-16 py-8 flex justify-between items-center border-b-2 text-blue-500 w-full"
    >
      <div
        class="flex items-center gap-4"
      >
        <img
          alt=""
          height="50"
          src="planet.png"
          width="50"
        />
        <h1
          class="text-black text-lg"
        >
          Space Travelers Hub
        </h1>
      </div>
      <ul
        class="flex gap-4 border-b-slate-400"
      >
        <li
          class="link"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li
          class="link"
        >
          <a
            class=""
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li
          class="link"
        >
          <a
            class=""
            href="/myprofile"
          >
            My profile
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
